data/src/main/java/pl/edu/agh/papaya/security/UserContext.java

```diff
@@ -0,0 +1,37 @@
+package pl.edu.agh.papaya.security;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import pl.edu.agh.papaya.model.User;
+
+@Component
+public class UserContext {
+
+    private UserPrincipal getPrincipal() {
+        Object principal = getAuthentication().getPrincipal();
+
+        if (principal instanceof UserPrincipal) {
+            return (UserPrincipal) principal;
+        }
+
+        throw new UserNotAuthenticatedException("Invalid principal: " + principal + " (" + principal.getClass() + ")");
+    }
+
+    private Authentication getAuthentication() {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        if (authentication == null) {
+            throw new UserNotAuthenticatedException();
+        }
+
+        return authentication;
+    }
+
+    public User getUser() {
+        return getPrincipal().getUser();
+    }
+
+    public String getUserId() {
+        return String.valueOf(getUser().getId());
```

Comment:
Moment, a w sumie czemu tu jest `String.valueOf`?

Reply:
Bo jak już odejdziemy od biedaloginu i zintegrujemy jakiś prawdziwy system logowania to jest naprawdę duża szansa, że long nie wystarczy jako ID.

Reply:
Jasne

Ale to ta metoda, o którą padło pytanie wyżej: https://github.com/papaya-agh/papaya-backend/pull/19#discussion_r277562587 używa stringa wyłącznie dlatego, że tu zrobiliśmy `String.valueof`?

Reply:
Raczej powiedziałbym, że z tego samego powodu co opisałem wyżej :smiley: ewentualnie można na razie zmienić na longa a potem jak nie wystarczy to się wróci do stringa, chociaż założyłem że łatwiej jest zmienić stringa na longa niż na odwrót

Reply:
`return getUser().getId().toString()` ?

Reply:
~No wiem, rozumiem przyszłościowy powód, ale w kontekście łapania `NumberFormatException` i tak dalej - jakby to się stało, to znaczy, że coś poszło bardzo bardzo nie tak w scenariuszu `Long -> String -> Long`. Jak odejdziemy od longów, to tam nie będzie żadnego łapania wyjątków i w scenariuszu "ktoś podał id, które nie istnieje" dostanie pustą listę~

Zwrot akcji: w trakcie pisania powyższego akapitu dotarło do mnie znaczenie tych słów:

>Good point, ale ogólnie średnio żeby stąd leciało bad request, bo to nie web, i raczej powinna lecieć jakaś 500 bo wtedy jest zalogowany user o złym ID. Tylko że wtedy znowu przydałoby się sprawdzić czy user o ID numerycznym istnieje, bo wtedy mielibyśmy niekonsystentne zachowanie: user, który nie istnieje o ID numerycznym nie będzie mieć projektów, a o ID nienumerycznym zwróci 500. Więc w sumie nie wiem czy tego nie zostawić tak, ewentualnie uzupełnić dokumentację o opis tego zachowania. Co o tym myślisz?

I konkluzja jest taka, że się zgadzam